mono/mini/mini-posix.c

```diff
@@ -717,6 +721,10 @@ sampling_thread_func (void *data)
 			/* info should never be this thread as we're a tools thread. */
 			g_assert (mono_thread_info_get_tid (info) != mono_native_thread_id_get ());
 
+			if (!InterlockedRead (&info->profiler_signal_ack))
```

Comment:
No particular reason other than me being tired when I wrote it - will change it.